io.openems.edge.core/src/io/openems/edge/app/timeofusetariff/AwattarHourly.java

```diff
@@ -66,34 +96,19 @@ public AwattarHourly(@Reference ComponentManager componentManager, ComponentCont
 	}
 
 	@Override
-	protected ThrowingTriFunction<ConfigurationTarget, EnumMap<Property, JsonElement>, Language, AppConfiguration, OpenemsNamedException> appConfigurationFactory() {
+	protected ThrowingTriFunction<ConfigurationTarget, Map<Property, JsonElement>, Language, AppConfiguration, OpenemsNamedException> appPropertyConfigurationFactory() {
 		return (t, p, l) -> {
 
-			final var alias = this.getValueOrDefault(p, Property.ALIAS, this.getName(l));
-
-			final var ctrlEssTimeOfUseTariffDischargeId = this.getId(t, p,
-					Property.CTRL_ESS_TIME_OF_USE_TARIF_DISCHARGE_ID, "ctrlEssTimeOfUseTariffDischarge0");
-			final var timeOfUseTariffId = this.getId(t, p, Property.TIME_OF_USE_TARIF_ID, "timeOfUseTariff0");
-
-			// TODO ess id may be changed
-			var components = Lists.newArrayList(//
-					new EdgeConfig.Component(ctrlEssTimeOfUseTariffDischargeId, alias,
-							"Controller.Ess.Time-Of-Use-Tariff.Discharge", JsonUtils.buildJsonObject() //
-									.addProperty("ess.id", "ess0") //
-									.build()), //
-					new EdgeConfig.Component(timeOfUseTariffId, this.getName(l), "TimeOfUseTariff.Awattar",
-							JsonUtils.buildJsonObject() //
-									.build())//
-			);
-			return new AppConfiguration(components,
-					Lists.newArrayList(ctrlEssTimeOfUseTariffDischargeId, "ctrlBalancing0"));
-		};
-	}
+			final var alias = this.getString(p, l, Property.ALIAS);
+			final var timeOfUseTariffProviderId = this.getId(t, p, Property.TIME_OF_USE_TARIF_PROVIDER_ID);
```

Comment:
i usually separate the ids from the other variables and also put them at the top of the method